Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs

```diff
@@ -1639,7 +1639,7 @@ public bool HasFront3DCab
             {
                 var loco = this as MSTSLocomotive;
                 var i = (int)CabViewType.Front;
-                if (loco == null || loco.CabView3D == null) return false;
+                if (loco == null || loco.CabView3D == null || loco.CabView3D.CabViewType != CabViewType.Front) return false;
```

Comment:
Nearby in this file, there is a counterpart `HasRear3DCab` getter that presumably needs to be changed, too.

Reply:
Locomotives with only a front cab will have just one CabViewpoint, so HasRear3DCab will work as expected. The HasFrontCab and HasRearCab 2D counterparts follow the same logic.

Reply:
I see, thanks!